dev/io.openliberty.microprofile.metrics.internal.5.x.monitor_fat/fat/src/io/openliberty/microprofile/metrics/internal/monitor_fat/ComputedMetricsTest.java

```diff
@@ -0,0 +1,637 @@
+                    return true;
+                }
+            });
+            String authorization = "Basic " + Base64.getEncoder().encodeToString(
+                    ("theUser:thePassword").getBytes(StandardCharsets.UTF_8)); // Java
+            con.setRequestProperty("Authorization", authorization);
+            con.setRequestMethod("GET");
+
+            String sep = System.getProperty("line.separator");
+            String line = null;
+            StringBuilder lines = new StringBuilder();
+            BufferedReader br = new BufferedReader(
+                    new InputStreamReader(con.getInputStream()));
+
+            while ((line = br.readLine()) != null && line.length() > 0) {
+                if (!line.startsWith("#"))
+                    lines.append(line).append(sep);
+            }
+            Log.info(c, "getHttpsServlet", sURL);
+            return lines.toString();
+        } finally {
+            if (con != null)
+                con.disconnect();
+        }
+    }
+
+    private void checkForExpectedStrings(String metricsText, String[] expectedStrings) {
+        Log.info(c, "checkForExpectedStrings", "metricsText:\n" + metricsText);
+        String sep = System.getProperty("line.separator");
+        String[] metricArray = metricsText.split(sep);
+
+        for (String m : expectedStrings) {
+            if (!metricsText.contains(m)) {
+                Log.info(c, "checkForExpectedStrings", "Failed:\n" + metricsText);
+                Assert.fail("Did not contain expected string: " + m);
+            }
+
+            if (!isComputedMetricANumber(metricArray, m)) {
+                Assert.fail("The computed metric " + m + " is not numeric.");
+            }
+        }
+    }
+
+    private void checkForUnexpectedStrings(String metricsText,
+            String[] unexpectedStrings) {
+        Log.info(c, "checkForUnexpectedStrings", "metricsText:\n" + metricsText);
+
+        for (String m : unexpectedStrings) {
+            if (metricsText.contains(m)) {
+                Log.info(c, "checkForExpectedStrings", "Failed:\n" + metricsText);
+                Assert.fail("Contains unexpected string: " + m);
+            }
+        }
+    }
+
+    private boolean isComputedMetricANumber(String[] metricsArray, String computedMetric) {
+        for (String metric : metricsArray) {
+            if (metric.contains(computedMetric)) {
+                Log.info(c, "isComputedMetricANumber",
+                        "Found Computed Metric : " + metric);
+                // Gets the metricValue from the metricName
+                String metricValue = getMetricValueFromMetric(metric);
+                Log.info(c, "isComputedMetricANumber",
+                        "The metric value is : " + metricValue);
+                return isStringANumber(metricValue);
+            }
+        }
+        return false;
+    }
+
+    private String getMetricValueFromMetric(String metricString) {
```

Comment:
Could maybe just combine this and and the isStringANumber into one method.

Reply:
Done.